public/src/circuitElement.js

```diff
@@ -22,6 +22,7 @@ import { generateNodeName } from './verilogHelpers';
 export default class CircuitElement {
     constructor(x, y, scope, dir, bitWidth) {
         // Data member initializations
+        const Diodeflag = 0 ; 
```

Comment:
You are making changes to the base class here. That is unacceptable for this feature. Why was this needed?

Reply:
This is needed to resolve the issue of stack simulation error .

Reply:
TriState does everything that diode does and more. Refactor this PR similar to how TriState is.